### PR TITLE
Updated method signature for ListView#scrollTo

### PR DESCRIPTION
**Pull request by jordanpapaleo**
**Note: The test suite is throwing errors after a fresh fork and clone**

### ListView

- The method signature in ListView.scrollTo was not helpful and basically acted as a passthrough to ScrollView#scrollTo.  I thought it would be helpful to define the args here instead of having an array of any.
- I also thought it would be helpful to fix deprecated arguments before passing them to ScrollView#scrollTo.
- Since I fix the arguments here, the warning message in scrollTo will never be called, so I added a warning message here.
- Additionally I used a params object to be passed to ScrollView#scrollTo in order have a single exit point from the function instead of returning in each bool case.
- Lastly I updated the comments for this function

### ScrollView

- I updated the warning message for the deprecated args in ScrollView#scrollTo to clarify the method signature as the issue.  This message also matches the logging I added to ListView#scrollTo

## Test Plan (required)

Create a basic list ViewComponent with a touchable highlight to used to scroll the list.  Call the scroll to function with the deprecated params (y, x, animated) and with the current params ({x, y: animated})

There was not a unit test for ListView so I added one but I am not sure how to test the scrolling of it.

I hope this is good enough to get in :)

Please let me know if there is anything I can do to make it better.

### CODE REVIEW

File: Libraries/Lists/ListView/ListView.js [hunk 1]
```diff
@@ -259,13 +259,46 @@ var ListView = React.createClass({
...
+    y?: number | { x?: number, y?: number, animated?: boolean },
+    x?: number,
+    animated?: boolean
+  ) {
     if (this._scrollComponent && this._scrollComponent.scrollTo) {
-      this._scrollComponent.scrollTo(...args);
+      var scrollToProps = {};
+
+      // The correct params were passed so use them
+      if (typeof y === 'object') {
```
Comment (chirag04) [inline Libraries/Lists/ListView/ListView.js]: there is no need to add any logic here. this is delegating to the scrollview and you already have a warning there. am i missing something?

File: Libraries/Lists/ListView/__tests__/ListView-tests.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+'use strict';
+
+jest.disableAutomock();
+
+const React = require('React');
+const ReactTestRenderer = require('react-test-renderer');
+const ListView = require('ListView');
+const Text = require('Text');
+
+describe('ListView', () => {
```
Comment (chirag04) [inline Libraries/Lists/ListView/__tests__/ListView-tests.js]: i would rather add tests to scrollview for this. listiview is delegating to scrollview#scrollTo

File: Libraries/Lists/ListView/ListView.js (hunk 1 above)
Comment (jordanpapaleo) [inline Libraries/Lists/ListView/ListView.js]: hi - it was  more that when you look at this function, the params were ..args array of anything.  My goal was to make this method signature clear and concise.

Comment (jordanpapaleo) [inline Libraries/Lists/ListView/ListView.js]: It also seemed good update/fix the args closer to the source of the incorrect args being set.

Comment (jordanpapaleo) [inline Libraries/Lists/ListView/ListView.js]: @chirag04 Let me know if you think this PR is not helpful and I will close it.  Again my goal was making it easier for developers to read the source :)

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/updated-method-signature-for-listviewscrollto

**Comment by hramos** [comment]
Closed in error.

**Comment by jordanpapaleo** [comment]
Any update on this?

Thanks!  jordan

**Comment by hramos** [comment]
I think I can take a look at this one tomorrow.

**Comment by hramos** [comment]
@chirag04 after looking at the comments it seems like you have some opinions on this. Do you want to take another look?

**Comment by jordanpapaleo** [comment]
Thanks for trying :)

Sent from my smallest computer

On Sep 3, 2017, at 10:04 AM, Facebook Community Bot <notifications@github.com> wrote:

@jordanpapaleo I tried to find reviewers for this pull request and wanted to ping them to take another look. However, based on the blame information for the files in this pull request I couldn't find any reviewers. This sometimes happens when the files in the pull request are new or don't exist on master anymore. Is this pull request still relevant? If yes could you please rebase? In case you know who has context on this code feel free to mention them in a comment (one person is fine). Thanks for reading and hope you will continue contributing to the project.

**Comment by chirag04** [comment]
@hramos ListView, FlatList, VirtualizedList etc are all wrapper on top of scrollview and they all delegate to `ScrollView.scrollTo`. I think it's weird to add the warning in ListView. adding to scrollview will benefit flatList etc.

anyway, i don't feel strongly about it. cc @sahrens

**Pull request closed without merging**